src/tool/xmeta/cpp/models/class_or_interface_model.h

```diff
@@ -34,46 +43,36 @@ namespace xlang::xmeta
             return m_events;
         }
 
-        void add_interface_base_ref(std::string_view const& interface_base_ref)
-        {
-            m_interface_base_refs.emplace_back(std::string(interface_base_ref));
-        }
+        void add_interface_base_ref(std::string_view const& interface_base_ref);
 
-        void add_interface_base_ref(size_t index, std::shared_ptr<interface_model> interface_base_ref)
-        {
-            assert(index < m_interface_base_refs.size());
-            m_interface_base_refs[index].resolve(interface_base_ref);
-        }
+        compilation_error add_member(std::shared_ptr<method_model> const& member);
```

Comment:
are add_member overloads called during parsing? If so, why would it return a compilation_error?

Reply:
I used compilation_error to return the message that add_member can't add the member since the member with the same name already exist in the model. They are not called during parsing. They are called when we are walking the AST and putting them into the models.